lib/onelogin/ruby-saml/metadata.rb

```diff
@@ -85,6 +82,13 @@ def generate(settings)
         #  <md:XACMLAuthzDecisionQueryDescriptor WantAssertionsSigned="false" protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol"/>
 
         meta_doc << REXML::XMLDecl.new("1.0", "UTF-8")
+
+        # embed signature
+        if settings.security[:metadata_signed] && settings.private_key && settings.certificate && settings.security[:embed_sign]
```

Comment:
Good point. I just pushed up a change that removes the check for `security[:embed_sign]`.